lib/components/platedhole.ts

```diff
@@ -36,10 +36,22 @@ export interface PillPlatedHoleProps
   portHints?: PortHints
 }
 
+export interface CircularHoleWithSquarePadProps
+  extends Omit<PcbLayoutProps, "pcbRotation" | "layer"> {
+  name?: string
+  shape: "circular_hole_with_square_pad"
+  holeSideLength: number | string
+  outerSideLength: number | string
+  holeShape: "circle"
+  padShape: "square"
+  portHints?: PortHints
+}
+
 export type PlatedHoleProps =
   | CirclePlatedHoleProps
   | OvalPlatedHoleProps
   | PillPlatedHoleProps
+  | CircularHoleWithSquarePadProps
```

Comment:
The same for this should be something like this `CircularHoleWithSquarePlatedProps` which follows the similar naming pattern with other props name